js/algoliasearch/click_conversion_analytics.js

```diff
@@ -43,47 +54,48 @@ algoliaBundle.$(function ($) {
 	// "algoliaConfig.ccAnalytics.orderedProductIds" are set only on checkout success page
 	if (algoliaConfig.ccAnalytics.conversionAnalyticsMode === 'place_order'
 		&& algoliaConfig.ccAnalytics.orderedProductIds.length > 0) {
-		$.each(algoliaConfig.ccAnalytics.orderedProductIds, function (i, objectId) {
-			trackConversion(objectId);
-		});
+		trackConversion(algoliaConfig.indexName + "_products", algoliaConfig.ccAnalytics.orderedProductIds);
 	}
 
 });
 
-algolia.registerHook('beforeInstantsearchInit', function (instantsearchOptions) {
-	instantsearchOptions.searchParameters['clickAnalytics'] = true;
+var analyticsHelper = {};
 
-	return instantsearchOptions;
+algolia.registerHook('beforeAutocompleteSources', function(sources) {
+	analyticsHelper.sources = sources;
+	return sources;
 });
 
 algolia.registerHook('beforeInstantsearchStart', function (search) {
 	search.once('render', function() {
-		AlgoliaAnalytics.initSearch({
-			getQueryID: function() {
-				return search.helper.lastResults && search.helper.lastResults._rawResults[0].queryID
-			}
-		});
+		analyticsHelper.getLastResults = function () {
+			return search.helper.lastResults;
+		}
 	});
-
 	return search;
 });
 
+algolia.registerHook('beforeInstantsearchInit', function (instantsearchOptions) {
+	instantsearchOptions.searchParameters['clickAnalytics'] = true;
+	return instantsearchOptions;
+});
 
-function trackClick(objectID, position, queryId) {
+function trackClick(indexName, objectID, position, queryId) {
 	var clickData = {
-		objectID: objectID.toString(),
-		position: parseInt(position)
+		index: indexName,
+		eventName: "Clicked item",
+		objectIDs: [objectID.toString()],
+		positions: [parseInt(position)],
+		queryID: queryId
 	};
 
-	if (queryId) {
-		clickData.queryID = queryId;
-	}
-
-	AlgoliaAnalytics.click(clickData);
+	AlgoliaAnalytics.clickedObjectIDsAfterSearch(clickData);
 }
 
-function trackConversion(objectID) {
-	AlgoliaAnalytics.conversion({
-		objectID: objectID.toString()
+function trackConversion(indexName, objectIDs) {
+	AlgoliaAnalytics.convertedObjectIDsAfterSearch({
+		index: indexName,
+		eventName: "Conversion",
```

Comment:
if this is for perso, we should pick a better Event Name.
@tvweinstock any recommandations or patterns that emerged for perso event names?

Reply:
@tkrugg is this for perso? I thought that `convertedObjectIDsAfterSearch ` was for clickAnalytics and `convertedObjectIDs` was for personalisation.

Reply:
@tkrugg @tvweinstock  this PR was specifically made to update the search-insights.js and click_conversion_analytics.js for the insights update. We're not adding personalisation in as it's more involved with fetching customerIDs and what not.

Reply:
alright, sorry I got a little confused because I thought we were doing this specifically for allowing perso. Because if I read this correctly, click analytics already work with the old implem. Am I forgetting something?

Reply:
@bsuravech so I talked about this with @tvweinstock and apparently we need to support perso as well. Is this something we'd want to tackle in another PR ?